best-practices/README.md

```diff
@@ -0,0 +1,71 @@
+Best Practices
+=============
+
+The following are locally agreed upon best practices. As often as possible,
+these will inherit from practices within the larger open source development
+community at large. These best practices may, and likely will, change in the
+future. But that change will be incorporated via group consensus and documented
+here.
+
+* [Documentation](documentation.md)
+* [Project README's](project_readme.md)
+* [Deployment](deployment.md)
+
+
+General
+-------
+
+* These are not to be blindly followed; strive to understand these and ask
+  when in doubt.
+* Don't duplicate the functionality of a built-in library.
+* Don't swallow exceptions or "fail silently."
+* Don't write code that guesses at future functionality.
+* Exceptions should be exceptional.
+* [Keep the code simple].
+
+[Keep the code simple]: http://www.readability.com/~/ko2aqda2
+
+Object-Oriented Design
+----------------------
+
+* Avoid global variables.
+* Avoid long parameter lists.
+* Limit collaborators of an object (entities an object depends on).
+* Limit an object's dependencies (entities that depend on an object).
+* Prefer composition over inheritance.
+* Prefer small methods. Between one and ten lines is best.
+* Prefer small classes with a single, well-defined responsibility. When a
+  class exceeds 100 lines, it may be doing too many things.
+* [Tell, don't ask].
+
+[Tell, don't ask]: http://robots.thoughtbot.com/post/27572137956/tell-dont-ask
+
+Ruby
+----
+
+* Follow the [Ruby Style Guide](https://github.com/bbatsov/ruby-style-guide)
+* Follow the [RSpec Guidelines with Ruby](http://betterspecs.org/)
+* Use rbenv and include a .ruby-version file to specify the Ruby version for the project
+* Avoid optional parameters. Does the method do too much?
+* Avoid monkey-patching.
+* Generate necessary [Bundler binstubs] for the project, such as `rake` and
+  `rspec`, and add them to version control.
+* Prefer classes to modules when designing functionality that is shared by
+  multiple models.
```

Comment:
Yes, I was debating removing this entirely. it may be out of scope to enforce a design pattern in this manner, and it's true that currently we take the modules approach. Perhaps delete the line altogether?